caml_z.c

```diff
@@ -1228,6 +1319,31 @@ CAMLprim value ml_z_fits_int32(value v)
   }
 }
 
+CAMLprim value ml_z_fits_int32_unsigned(value v)
+{
+  intnat x;
+  Z_MARK_OP;
+  Z_CHECK(v);
+  if (Is_long(v)) {
+    x = Long_val(v);
+    if (x < 0) return Val_false;
+#ifdef ARCH_SIXTYFOUR
+    if (x >= Z_HI_UINT32) return Val_false;
+#endif
+    return Val_true;
+  }
+  else {
+    Z_DECL(v);
+    Z_MARK_SLOW;
+    Z_ARG(v);
+    if (sign_v || size_v > 1) return Val_false;
+    if (!size_v) return Val_true;
+    x = *ptr_v;
+    if ((uintnat)x >= Z_HI_INT32) return Val_false;
```

Comment:
This doesn't feel right.  `Z_HI_INT32` is an upper bound for a signed 32-bit integer.  Also, going through a signed intnat feels unnecessary.  Suggestion:
```suggestion
#ifdef ARCH_SIXTYFOUR
    if (*ptr_v >= Z_HI_UINT32) return Val_false;
#endif
```